### PR TITLE
feat(btc-server): Sync entire wallet state

### PR DESCRIPTION
**Pull request by scottmillner**
this pr wires up the wallet state sync ability but does not "turn it on": there's no task that invokes `sync_wallet_state()`

### CODE REVIEW

File: bin/reth/src/commands/poa/mod.rs [hunk 1]
```diff
@@ -839,6 +841,26 @@ impl<Ext: clap::Args + fmt::Debug> PoaNodeCommand<Ext> {
             //         healthcheck_task.expect("health check task exists").start_task().await;
             //     }),
             // );
+
+            // just for local testing purposes
```
Comment (rwlockbg) [inline bin/reth/src/commands/poa/mod.rs]: maybe remove this fro the repo if it does not go into prod ?

File: crates/consensus/authority/src/frost_task.rs [hunk 2]
```diff
@@ -323,13 +400,29 @@ where
...
-                    PeerMessageResponse::Utxo(mut response) => {
+                    PeerMessageResponse::WalletState(mut response) => {
+                        // Only handle response if it has no state: responses with state are also
+                        // sent to WalletStateSyncEngine::sync_wallet_state
+                        // which updates the wallet state. This code block
+                        // handles sending our wallet state to a peer
+                        //
+                        // TODO: create separate messages for asking for wallet state and sending
+                        // wallet state
+                        if !response.utxos.is_empty() ||
```
Comment (rwlockbg) [inline crates/consensus/authority/src/frost_task.rs]: I think we might do some sort of validation func here that wraps all conditions together ?

File: crates/net/network/src/frost/messages.rs [hunk 3]
```diff
@@ -551,6 +613,31 @@ impl FrostProtoMessage {
 
                 FrostProtoMessageKind::Healthcheck(HealthcheckRequest { sender, receiver })
             }
+            FrostProtoMessageId::WalletState => {
```
Comment (rwlockbg) [inline crates/net/network/src/frost/messages.rs:616]: Further down we have some ser/deser tests for the other message types, maybe we could add the WalletState to them too just for sanity and completeness?

**Review by rwlockbg — COMMENTED** [review]
Nice work! In fact, its the old utxo sync that we had but with the pending pegouts extended. I wonder if we should create some integration tests to confirm it all works or leave this as a separate PR ?

**Comment by scottmillner** [comment]
> Nice work! In fact, its the old utxo sync that we had but with the pending pegouts extended. I wonder if we should create some integration tests to confirm it all works or leave this as a separate PR ?

i was planning to do this is another pr since i want to stop work on this temporarily and focus on async pegouts

File: crates/consensus/authority/src/frost_task.rs (hunk 2 above)
Comment (scottmillner) [reply crates/consensus/authority/src/frost_task.rs]: done!

File: crates/net/network/src/frost/messages.rs (hunk 3 above)
Comment (scottmillner) [reply crates/net/network/src/frost/messages.rs:616]: done!

Comment (rwlockbg) [reply crates/net/network/src/frost/messages.rs:616]: nice!

File: crates/consensus/authority/src/frost_task.rs (hunk 2 above)
Comment (rwlockbg) [reply crates/consensus/authority/src/frost_task.rs]: nice!

File: bin/reth/src/commands/poa/mod.rs (hunk 1 above)
Comment (rwlockbg) [reply bin/reth/src/commands/poa/mod.rs]: thx

**Review by rwlockbg — APPROVED** [review]

**Pull request merged**